src/mavo.js

```diff
@@ -177,7 +177,13 @@ var _ = self.Mavo = $.Class({
 
 		this.permissions.can("login", () => {
 			// We also support a URL param to trigger login, in case the user doesn't want visible login UI
-			if (Mavo.Functions.url("login") !== null && this.index == 1 || Mavo.Functions.url(this.id + "-login") !== null) {
+			if (Mavo.Functions.url("login") !== null && this.index === 1) {
+				// Remove ?login from url
+                const newUrl = Mavo.Functions.url().replace("?login", "");
+				window.history.pushState("", "", newUrl);
+				this.primaryBackend.login();
```

Comment:
So you are removing the URL param before login actually happens. Is that a conscious decision? If login fails, do we want the URL param there or not?

Reply:
Yes this was a conscious decision on my part. At the moment if I navigate to a Mavo app with a ?login parameter and ignore the login popup, Mavo doesn't ask me to log in again. I figured it made more sense to ask once to login and redirect to the main site regardless of the login success/failure.